src/main/java/com/urlparser/matcher/Matcher.java

```diff
@@ -0,0 +1,73 @@
+package com.urlparser.matcher;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+/**
+ * This Matcher compare Iterable<T> with Iterable<T> for equals.
```

Comment:
@undsett 
1. `compares`
2. Lets remove <T> since the Matcher is not generic, there is no `T` definition